Assets/Scripts/BlockScripts/nextLetter.cs

```diff
@@ -3,36 +3,27 @@
 
 public class NextLetter : MonoBehaviour
 {
-    // Start is called before the first frame update
-    void Start()
-    {
-        //alphabet = createAlphabet(language);
-    }
-
-    // Update is called once per frame
-    void Update()
-    {
-        
-    }
 
-    public static string getNextLetter(List<Letter> alphabet, int totalWeight){
-
-        // totalWeight is the sum of all alphabet' weight
+    public static char GetNextLetter(List<Letter> alphabet){
+        int totalWeight = 0; 
+        foreach(var letter in alphabet)
+        {
+            totalWeight += letter.Value;
+        }
 
         int randomNumber = Random.Range(0, totalWeight);
 
-        string selectedLetter = null;
-        foreach(Letter letter in alphabet)
+        foreach(var letter in alphabet)
         {
-            if (randomNumber < letter.weight)
+            if (randomNumber < letter.Value)
             {
-                return selectedLetter;
+                return letter.Label;
             }
 
-            randomNumber = randomNumber - letter.weight;
+            randomNumber -= letter.Value;
         }
 
-        return "";
+        return 'A';
```

Comment:
```suggestion
        return string.empty;
```
Easier to see if there is a bug

Reply:
the method returns a character

Reply:
would you have any complaints with `return '\0'`

Reply:
NVM, A is fine then

Reply:
```suggestion
            public static char GetNextLetter(List<Letter> alphabet)
    {
        int totalWeight = 0;
        char val = new char();

        foreach (var letter in alphabet)
        {
            totalWeight += letter.Value;
        }

        int randomNumber = Random.Range(0, totalWeight);

        foreach (var letter in alphabet)
        {
            if (randomNumber < letter.Weight)
            {
                val = letter.Value;
            }
            else
            {
                val = '\0';
            }
            randomNumber -= letter.Value;
        }

        return val;
    }
```

Reply:
Take a look at the new code pushed up, ill keep convo open for now